### PR TITLE
Improve user permission detection on Windows

### LINKED ISSUE #4503: Hyper-V provider requires user be an admin instead of requiring user has Hyper-V permissions
In order to interact with Hyper-V the user needs specific permissions (normally only provided to administrators) however these permissions can be granted to other users. I have done so with my account and can access hyper-v both from the Hyper-V manager as well as through powershell without elevating to admin. Unfortunately Vagrant requires me to be an admin. It would be much better if Vagrant would verify that the user has access to Hyper-V before requiring elevation.

### PR DESCRIPTION
**Pull request by lukebakken**
Improve detection of admin and hyper-v admin rights via using SIDs and powershell

Fixes #4503 and improves on PR #6662

Tested using the following:
- Windows 10
- `ruby 2.2.5p319 (2016-04-26 revision 54774) [x64-mingw32]`
- Vagrant box: `hashicorp/precise64 (hyperv, 1.1.0)`

### CODE REVIEW

**Comment by lukebakken** [comment]
Hm, looks like the Travis build failed due to a bundler version issue.

**Comment by lukebakken** [comment]
@shamakhov - if you can, give this branch a try. It should detect that you are a Hyper-V admin no matter what language you're using.

**Comment by chrisroberts** [comment]
@lukebakken if you want to rebase on master it should clear the travis issue up. thanks!

**Comment by lukebakken** [comment]
@chrisroberts - rebase done, let's see if Travis is happy now :smile: :fish_cake:

File: lib/vagrant/util/platform.rb [hunk 1]
```diff
@@ -42,27 +43,17 @@ def windows?
...
           @_windows_admin = -> {
-            begin
-              Win32::Registry::HKEY_USERS.open("S-1-5-19") {}
-
-              # The above doesn't seem to be 100% bullet proof. See GH-5616.
-              return (`reg query HKU\\S-1-5-19 2>&1` =~ /ERROR/).nil?
-            rescue Win32::Registry::Error
-              return false
-            end
+            ps_cmd = "[System.Security.Principal.WindowsIdentity]::GetCurrent().Groups | ForEach-Object { if ($_.Value -eq 'S-1-5-19'){ Write-Host 'true'; break }}"
```
Comment (chrisroberts) [inline lib/vagrant/util/platform.rb]: This modification breaks the functionality on older versions of windows. We need to either revert it back to the original form, or merge the two and allow fallback to original check implementation.

Comment (lukebakken) [inline lib/vagrant/util/platform.rb]: Couple questions -

* What was the failure?
* Which version did you notice it failing?

Thanks

Comment (mwrock) [inline lib/vagrant/util/platform.rb]: This also works as far back as win 7/2008R2 powershell v2
```
    $identity  = [System.Security.Principal.WindowsIdentity]::GetCurrent()
    $principal = New-Object System.Security.Principal.WindowsPrincipal( $identity )
    if($principal.IsInRole( [System.Security.Principal.WindowsBuiltInRole]::Administrator )) { Write-Host 'true' }
```

Comment (lukebakken) [inline lib/vagrant/util/platform.rb]: @mwrock - I like your solution better than mine, but I'm curious if my code failed in your environment. I suspect @chrisroberts is interested in support even further back.

Comment (tsafin) [inline lib/vagrant/util/platform.rb]: There is no much to go farther: Hyper-V was initially released 1 year before 2008R2, with Windows 2008, and there was optional PowerShell V1 component available. 2008R2 was released with PowerShell V2, and real question here is - how different syntaxes of PowerShell V1 and V2 are here, for these particular lines. I guess not (but I'm not that much profecient in PowerShell).

**Pull request merged**